RUFAS/input_manager.py

```diff
@@ -128,6 +128,296 @@ def start_data_processing(self, metadata_path: Path, eager_termination: bool = T
+        self.om.add_error(
+            "Runtime metadata load failure",
+            "No metadata is currently loaded into the InputManager.",
+            info_map,
+        )
+        return False
+
+    def _get_runtime_metadata_map(self, info_map: Dict[str, Any]) -> Dict[str, Any] | None:
+        try:
+            return self.get_metadata("runtime_metadata")
+        except KeyError:
+            self.om.add_error(
+                "Runtime metadata not configured",
+                "No runtime metadata configuration found in the active metadata document.",
+                info_map,
+            )
+            return None
+
+    def _get_runtime_metadata_reference(
+        self, runtime_metadata_map: Dict[str, Any], metadata_key: str, info_map: Dict[str, Any]
+    ) -> Dict[str, Any] | None:
+        metadata_reference = runtime_metadata_map.get(metadata_key)
+        if metadata_reference is None:
+            self.om.add_error(
+                "Runtime metadata key missing",
+                f"Runtime metadata key '{metadata_key}' not found in configuration.",
+                info_map,
+            )
+            return None
+        return metadata_reference
+
+    def _get_runtime_metadata_path(
+        self, metadata_reference: Dict[str, Any], metadata_key: str, info_map: Dict[str, Any]
+    ) -> Path | None:
+        metadata_path_value = metadata_reference.get("path")
+        if metadata_path_value is None:
+            self.om.add_error(
+                "Runtime metadata path missing",
+                f"Runtime metadata entry '{metadata_key}' is missing a path attribute.",
+                info_map,
+            )
+            return None
+        return Path(metadata_path_value)
+
+    def _load_runtime_metadata_contents(self, metadata_path: Path, info_map: Dict[str, Any]) -> Dict[str, Any] | None:
+        try:
+            return self._load_runtime_metadata_document(metadata_path)
+        except Exception as exc:
+            self.om.add_error(
+                "Runtime metadata load failure",
+                f"Failed to load runtime metadata from {metadata_path}: {exc}",
+                info_map,
+            )
+            return None
+
+    def _validate_runtime_metadata(self, runtime_metadata: Dict[str, Any], info_map: Dict[str, Any]) -> bool:
+        is_valid, message = self.data_validator.validate_metadata(
+            runtime_metadata, VALID_INPUT_TYPES, ADDRESS_TO_INPUTS
+        )
+        if is_valid:
+            return True
+        self.om.add_error("Runtime metadata validation failed", message, info_map)
+        self.om.route_logs(self.data_validator.event_logs)
+        return False
+
+    def _extract_runtime_files(
+        self, runtime_metadata: Dict[str, Any], metadata_path: Path, info_map: Dict[str, Any]
+    ) -> Dict[str, Any] | None:
+        runtime_files = runtime_metadata.get(ADDRESS_TO_INPUTS, {})
+        if isinstance(runtime_files, dict):
+            return runtime_files
```

Comment:
```suggestion
        runtime_files = runtime_metadata.get(ADDRESS_TO_INPUTS, None)
        if isinstance(runtime_files, dict):
            return runtime_files
```
so that when no results are found, we would not go into the if statement.